flink-test-utils-parent/flink-test-utils/src/main/java/org/apache/flink/networking/NetworkFailureHandler.java

```diff
@@ -48,6 +48,9 @@
     private static final Logger LOG = LoggerFactory.getLogger(NetworkFailureHandler.class);
     private static final String TARGET_CHANNEL_HANDLER_NAME = "target_channel_handler";
 
+    // They key set of this map contains the channels that are being closed. This is needed to
+    // prevent closeOnFlush() from being called recursively.
+    private static final Map<Channel, Channel> channelsBeingClosed = new ConcurrentHashMap<>();
```

Comment:
How about we use:

```
    private static final Set<Channel> channelsBeingClosed = ConcurrentHashMap.newKeySet();
```

Reply:
Thanks @dawidwys. This is nice to know. I have updated PR as suggested.